stl/inc/ranges

```diff
@@ -6821,6 +6822,496 @@ namespace ranges {
+            }
+
+            _NODISCARD_FRIEND constexpr auto iter_move(const _Iterator& _Itr) noexcept(
+                (noexcept(_RANGES iter_move(_STD declval<const iterator_t<_Maybe_const<_IsConst, _ViewTypes>>&>()))
+                    && ...)
+                && (is_nothrow_move_constructible_v<
+                        range_rvalue_reference_t<_Maybe_const<_IsConst, _ViewTypes>>> && ...)) {
+                return _Tuple_transform(_RANGES iter_move, _Itr._Current);
+            }
+
+            friend constexpr void iter_swap(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+                (noexcept(_RANGES iter_swap(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>(),
+                     _STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>()))
+                    && ...)) requires(indirectly_swappable<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>&&...) {
+                const auto _Swap_every_pair_lambda = [&_Lhs, &_Rhs ]<size_t... _Indices>(index_sequence<_Indices...>) {
+                    ((_RANGES iter_swap(_STD get<_Indices>(_Lhs._Current), _STD get<_Indices>(_Rhs._Current))), ...);
+                };
+
+                _Swap_every_pair_lambda(index_sequence_for<_ViewTypes...>{});
+            }
+        };
+
+        template <bool _IsConst>
+        class _Sentinel {
+        private:
+            using _Sentinel_tuple_type = tuple<sentinel_t<_Maybe_const<_IsConst, _ViewTypes>>...>;
+            /* [[no_unique_address]] */ _Sentinel_tuple_type _End;
+
+            friend zip_view;
+
+            constexpr explicit _Sentinel(_Sentinel_tuple_type _End) noexcept(
+                is_nothrow_copy_constructible_v<_Sentinel_tuple_type>) // strengthened
+                : _End(_End) {}
+
+            template <bool _IteratorConst>
+            using _Iterator_sentinel_difference_type =
+                common_type_t<range_difference_t<_Maybe_const<_IteratorConst, _ViewTypes>>...>;
+
+            template <bool _IteratorConst>
+            using _Iterator_type = typename zip_view<_ViewTypes...>::template _Iterator<_IteratorConst>;
+
+            template <bool _IteratorConst>
+            _NODISCARD static inline constexpr const auto& _Get_iterator_tuple(
+                const _Iterator_type<_IteratorConst>& _Itr) noexcept {
+                // TODO: This function shouldn't be necessary, but the MSVC throws a false
+                // C2248 error when doing _Lhs._Current directly within the definition of
+                // _Sentinel::operator==().
+                return _Itr._Current;
+            }
+
+            template <bool _IteratorConst>
+            static constexpr bool _Iterator_sentinel_equality_noexcept = noexcept(
+                _Zip_evaluate_iterator_sentinel_equality(_STD declval<const _Iterator_type<_IteratorConst>&>()._Current,
+                    _STD declval<_Sentinel_tuple_type>()));
+

```

Comment:
Neither does this constant (just ask for `noexcept(noexcept(_Zip_evaluate_iterator_sentinel_equality(blah, bloop)))`)